assemblers/org.osbuild.qemu

```diff
@@ -136,10 +136,23 @@ def main(tree, output_dir, options, loop_client):
     extra_args = []
 
     # raw and vdi don't suppport compression
-    if fmt not in ("raw", "vdi"):
+    if fmt not in ("raw", "vdi", "vpc"):
         extra_args.append("-c")
 
-    subprocess.run(["qemu-img", "convert", "-O", fmt, *extra_args, image, f"{output_dir}/{filename}"], check=True)
+    if fmt != "vpc":
+        subprocess.run(["qemu-img", "convert", "-O", fmt, *extra_args, image, f"{output_dir}/{filename}"], check=True)
+    else:
+        # Implementation of the resize algorithm from here:
+        # https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-generic
+        mb = 1024*1024
+        rounded_size = int((size/mb + 1) * mb)
+        subprocess.run(["qemu-img", "resize", image, rounded_size], check=True)
```

Comment:
Hm, I think we should simply require the size to be passed in correctly. If being 1MB algined is inherent to vpc, then do (vpc specific) a check above where we currently check for size being 512 bytes aligned. But if this is just Azure specific, I would not specify anything here, but rather rely on the caller (osbuild-composer) to provide the right value.

Reply:
Sure, size the size is passed as an option to the assembler, it would work as well.